astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1008 @@
+    } else {
+        result = set_double("r0", value, &r0);
+        if (result) return result;
+        if (r0 != self->x->r0) {
+            self->x->r0 = r0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_phi0(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else if (self->x->phi0 == UNDEFINED) {
+        Py_RETURN_NONE;
+    } else {
+        return get_double("phi0", self->x->phi0);
+    }
+}
+
+
+static int PyPrjprm_set_phi0(PyPrjprm* self, PyObject* value, void* closure)
+{
+    int result;
+    double phi0;
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        if (self->x->phi0 != UNDEFINED) {
+            self->x->phi0 = UNDEFINED;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    } else {
+        result = set_double("phi0", value, &phi0);
+        if (result) return result;
+        if (phi0 != self->x->phi0) {
+            self->x->phi0 = phi0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_theta0(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else if (self->x->theta0 == UNDEFINED) {
+        Py_RETURN_NONE;
+    } else {
+        return get_double("theta0", self->x->theta0);
+    }
+}
+
+
+static int PyPrjprm_set_theta0(PyPrjprm* self, PyObject* value, void* closure)
+{
+    int result;
+    double theta0;
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        if (self->x->theta0 != UNDEFINED) {
+            self->x->theta0 = UNDEFINED;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    } else {
+        result = set_double("theta0", value, &theta0);
+        if (result) return result;
+        if (theta0 != self->x->theta0) {
+            self->x->theta0 = theta0;
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        }
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_get_pv(PyPrjprm* self, void* closure)
+{
+    int k;
+    Py_ssize_t size = PVN;
+    double *pv;
+    PyObject* pv_array;
+
+    if (is_prj_null(self)) return NULL;
+
+    pv_array = (PyArrayObject*) PyArray_SimpleNew(1, &size, NPY_DOUBLE);
+    if (pv_array == NULL) return NULL;
+    pv = (double*) PyArray_DATA(pv_array);
+
+    for (k = 0; k < PVN; k++) {
+        if (self->x->pv[k] == UNDEFINED) {
+            pv[k] = (double) NPY_NAN;
+        } else {
+            pv[k] = self->x->pv[k];
+        }
+    }
+
+    return pv_array;
+}
+
+
+static int PyPrjprm_set_pv(PyPrjprm* self, PyObject* value, void* closure)
```

Comment:
Is there a test for this method?
Should I be able to set all `pv` values to `NaN`?
Regardless, I don't understand the result - some are set, some not.

Reply:
> Is there a test for this method?

Of course! See https://github.com/astropy/astropy/blob/a4875a1beb2eff084652f8eb99fb8f178759dce8/astropy/wcs/tests/test_prjprm.py#L114

> Should I be able to set all pv values to NaN?

I do not know what to say. I do not see why not even if it doesn't make much sense (just like setting all PVs to 1). I fixed a bug and added a test to `test_prjprm_pv()` to test this case.

> Regardless, I don't understand the result - some are set, some not.

Yes, you caught a bug. You should be able to set all PVs to `nan` as `pv = 30 * [np.nan]`. After next commit you should be able to set using `numpy.ndarray`.